core/src/main/java/org/kohsuke/stapler/export/ExportConfig.java

```diff
@@ -1,66 +1,146 @@
+        this.skipIfFail = skipIfFail;
+    }
 
     /**
-     * Interceptor used to check if given model object or it's property should be included in serialization
+     * If true, output will be indented to make it easier for humans to understand.
      */
-    private ExportInterceptor exportInterceptor = ExportInterceptor.DEFAULT;
-
-    private boolean skipIfFail = false;
-
     public boolean isPrettyPrint() {
         return prettyPrint;
     }
 
     /**
-     * If true, output will be indented to make it easier for humans to understand.
+     * Controls the behaviour of the class attribute to be produced.
      */
-    public ExportConfig withPrettyPrint(boolean prettyPrint) {
-        this.prettyPrint = prettyPrint;
-        return this;
-    }
-
     public ClassAttributeBehaviour getClassAttribute() {
         return classAttribute;
     }
 
     /**
      * Controls the behaviour of the class attribute to be produced.
      */
-    public ExportConfig withClassAttribute(ClassAttributeBehaviour cab) {
-        if (cab==null)  throw new NullPointerException();
-        this.classAttribute = cab;
-        return this;
+    public ClassAttributeBehaviour getClassAttributeBehaviour(){
+        return this.classAttribute;
     }
 
+    /**
+     * Gives {@link ExportInterceptor}. Always non-null.
+     */
     public ExportInterceptor getExportInterceptor() {
         return exportInterceptor;
     }
 
-    public ExportConfig withExportInterceptor(ExportInterceptor interceptor){
-        this.exportInterceptor = interceptor;
-        return this;
+    /**
+     *  Tells whether to skip serialization failure.
+     */
+    public boolean isSkipIfFail() {
+        return skipIfFail;
     }
 
-    public ExportConfig withSkipIfFail(boolean skipIfFail){
-        this.skipIfFail = skipIfFail;
-        return this;
+    /**
+     * Gives {@link Flavor}. Always non-null.
+     */
+    public Flavor getFlavor() {
+        return flavor;
     }
 
-    public boolean isSkipIfFail() {
-        return skipIfFail;
+    /**
+     * {@link ExportConfig} builder
+     */
+    public static class Builder{
+        private  boolean prettyPrint;
+
+        private  ClassAttributeBehaviour classAttribute = ClassAttributeBehaviour.IF_NEEDED;
+
+        private  ExportInterceptor exportInterceptor = ExportInterceptor.DEFAULT;
+
+        private  boolean skipIfFail = false;
+
+        private  final Flavor flavor;
+
+        public Builder(Flavor flavor){
+            this.flavor = flavor;
+        }
+
+        /**
+         * Turn on or off pretty printing of serialized data.
+         *
+         * @param prettyPrint default false.
+         *
+         * @return {@link Builder} instance
+         */
+        public Builder prettyPrint(boolean prettyPrint){
+            this.prettyPrint = prettyPrint;
+            return this;
+        }
+
+        /**
+         * If true serialization error will be ignored
```

Comment:
I am still a bit uneasy about this: it feels like rather than a hard-coded flag, this should just be an aspect of the behavior of `ExportInterceptor`.

Now currently `ExportInterceptor` is only asked to `getValue` on a single attribute, which is too late to defend against, say, an `Object` which is neither an `@ExportedBean` nor a `CustomExportedBean`. And the handling of those cases is hard-coded in Stapler core and not customizable—except via this flag.

So perhaps that means that the design of `ExportInterceptor` is wrong—too narrowly conceived. Rather than being asked to handle individual attributes, it should be asked to deal with the entire “bean”. The default implementation would check for `CustomExportedBean` and do a replacement if indicated; then fail if not `@ExportedBean`; then produce a `Model`, iterate its properties, and tell the `DataWriter` to create an object with all those properties. A custom implementation could be more or less lenient about various error conditions, or do something else entirely when encountering certain bean types (perhaps delegating to the default interceptor in other cases).

@kohsuke may want to weigh in here.